java/org/contikios/cooja/contikimote/ContikiMoteType.java

```diff
@@ -230,7 +231,8 @@ public Mote generateMote(Simulation simulation) {
   public boolean configureAndInit(Container parentContainer, Simulation simulation,
                                   boolean visAvailable) throws MoteTypeCreationException {
     myConfig = simulation.getCooja().getProjectConfig().clone();
-
+    String output_dir = Cooja.getExternalToolsSetting("PATH_CONTIKI_NG_BUILD_DIR", "");
```

Comment:
If the `PATH_CONTIKI_NG_BUILD_DIR` is missing in a configuration, I believe `output_dir` will be an empty string and Cooja might try to read/write files in the root directory (`/`). Even if the configuration always should exist I think it would be safer to always specify a default directory name like `build/cooja` instead of an empty string.

```suggestion
    String output_dir = Cooja.getExternalToolsSetting("PATH_CONTIKI_NG_BUILD_DIR", "build/cooja");
```

Reply:
@nfi changed across the board